galsim/chromatic.py

```diff
@@ -19,45 +19,89 @@
 
-    This class primarily serves as the base class for chromatic subclasses, including Chromatic,
-    ChromaticSum, and ChromaticConvolution.  See the docstrings for these classes for more details.
+    This class primarily serves as the base class for chromatic subclasses.  See the docstrings for
+    subclasses for more details.
 
     Initialization
     --------------
 
-    A ChromaticObject can also be instantiated directly from an existing GSObject.
-    In this case, the newly created ChromaticObject will act nearly the same way the original
-    GSObject works, except that it has access to the ChromaticObject methods described below;
-    especially expand(), dilate() and shift().
-
-    @param gsobj  The GSObject to be chromaticized.
+    A ChromaticObject can be instantiated from an existing GSObject by applying a
+    wavelength-dependent transformation.  E.g.,
+
+    >>> gsobj = galsim.Gaussian(fwhm=1)
+    >>> chrom_obj = gsobj.dilate(lambda wave: 1.01**wave)
+
+    In this and similar cases, the univariate object passed to the transformation method (`dilate`
+    above) is a python callable that accepts wavelength in nanometers and returns whatever type the
+    transformation method normally accepts (so an int or float above).
+
+    Another way to instantiate a ChromaticObject from a GSObject is to multiply by an SED.  This can
+    be useful to consistently generate the same galaxy observed through different filters, or, with
+    ChromaticSum, to construct multi-component galaxies, each component with a different SED.  For
+    example, a bulge+disk galaxy could be constructed:
+
+    >>> bulge_SED = user_function_to_get_bulge_spectrum()
+    >>> disk_SED = user_function_to_get_disk_spectrum()
+    >>> bulge_mono = galsim.DeVaucouleurs(half_light_radius=1.0)
+    >>> disk_mono = galsim.Exponential(half_light_radius=2.0)
+    >>> bulge = bulge_mono * bulge_SED
+    >>> disk = disk_mono * disk_SED
+    >>> gal = bulge + disk
+
+    The SEDs above describe the flux density in photons/nm/cm^2/s of an object, possibly normalized
+    with either the sed.withFlux(bandpass) or sed.withMagnitude(bandpass) methods (see the
+    docstrings in the SED class for details about these and other normalization options).  Note that
+    the `flux` attribute of the multiplied GSObject is accounted for in the newly constructed
+    ChromaticObject's normalization.  I.e., the following are equivalent:
+
+    >>> chrom_obj = (sed * 3.0) * gsobj
+    >>> chrom_obj2 = sed * (gsobj * 3.0)
+
+    Subclasses that instantiate a ChromaticObject directly also exist, such as ChromaticAtmosphere.
+    Even in this case, however, the underlying implementation always eventually wraps one or more
+    GSObjects.
+
+    ChromaticObjects can generally be sorted into two distinct types: those that represent galaxies
+    or stars and have dimensions of [photons/wavelength-interval/area/time/solid-angle], and those
```

Comment:
The area is referring to the collecting area of the telescope.  The solid angle is because we're modeling a surface brightness.  You need to integrate over both the collecting area and the galaxy solid angle (and exposure time and wavelength) to get number of collected photons.